### PR TITLE
Add com.rtosta.zapzap

### PR DESCRIPTION
**Pull request by rafatosta**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: com.rtosta.zapzap.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+app-id: com.rtosta.zapzap
+runtime: org.kde.Platform
+runtime-version: "6.2"
+sdk: org.kde.Sdk
+base: io.qt.qtwebengine.BaseApp
+base-version: "6.2"
+comand: "zapzap"
+finish-args:
+  - "--share=ipc"
+  - "--socket=fallback-x11"
```
Comment (hfiguiere) [inline com.rtosta.zapzap.yaml:10]: you have no wayland socket. so this won't work

File: com.rtosta.zapzap.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  - "--socket=pulseaudio"
+  - "--share=network"
+  - "--filesystem=xdg-download"
+  - "--device=dri"
+  - "--env=QTWEBENGINEPROCESS_PATH=/app/bin/QtWebEngineProcess"
+  - "--talk-name=org.kde.StatusNotifierWatcher"
+  - "--own-name=org.kde.*"
```
Comment (hfiguiere) [inline com.rtosta.zapzap.yaml:17]: do you really need to own this?

**Review by hfiguiere — CHANGES_REQUESTED** [review]

**Comment by hfiguiere** [comment]
bot, build com.rtosta.zapzap

File: com.rtosta.zapzap.yaml (hunk 2 above)
Comment (rafatosta) [reply com.rtosta.zapzap.yaml:17]: Really no. I will update

File: com.rtosta.zapzap.yaml (hunk 1 above)
Comment (rafatosta) [reply com.rtosta.zapzap.yaml:10]: With Wayland it is failing. I believe I should have just put x11. Correct?

**Comment by hfiguiere** [comment]
bot, build com.rtosta.zapzap

**Comment by nedrichards** [comment]
> flatpak run com.rtosta.zapzap
Traceback (most recent call last):
  File "/app/bin/sip-build", line 5, in <module>
    from sipbuild.tools.build import main
  File "/app/lib/python3.9/site-packages/sipbuild/__init__.py", line 25, in <module>
    from .abstract_builder import AbstractBuilder
  File "/app/lib/python3.9/site-packages/sipbuild/abstract_builder.py", line 26, in <module>
    from .configurable import Configurable
  File "/app/lib/python3.9/site-packages/sipbuild/configurable.py", line 24, in <module>
    from packaging.markers import Marker
ModuleNotFoundError: No module named 'packaging'

**Comment by rafatosta** [comment]
> > flatpak run com.rtosta.zapzap
> Traceback (most recent call last):
>   File "/app/bin/sip-build", line 5, in <module>
>     from sipbuild.tools.build import main
>   File "/app/lib/python3.9/site-packages/sipbuild/__init__.py", line 25, in <module>
>     from .abstract_builder import AbstractBuilder
>   File "/app/lib/python3.9/site-packages/sipbuild/abstract_builder.py", line 26, in <module>
>     from .configurable import Configurable
>   File "/app/lib/python3.9/site-packages/sipbuild/configurable.py", line 24, in <module>
>     from packaging.markers import Marker
> ModuleNotFoundError: No module named 'packaging'
> 

Very strange.Here it is working and on friends PC too. Shouldn't it have the same behavior regardless of location?

**Comment by rafatosta** [comment]
> > flatpak run com.rtosta.zapzap
> Traceback (most recent call last):
>   File "/app/bin/sip-build", line 5, in <module>
>     from sipbuild.tools.build import main
>   File "/app/lib/python3.9/site-packages/sipbuild/__init__.py", line 25, in <module>
>     from .abstract_builder import AbstractBuilder
>   File "/app/lib/python3.9/site-packages/sipbuild/abstract_builder.py", line 26, in <module>
>     from .configurable import Configurable
>   File "/app/lib/python3.9/site-packages/sipbuild/configurable.py", line 24, in <module>
>     from packaging.markers import Marker
> ModuleNotFoundError: No module named 'packaging'
> 

I did the test and the problem really appears when running via command line, but directly from the shortcut it works. I will insert the packaging module in yaml and test

**Comment by hfiguiere** [comment]
Are  you sure you launched the right one through the shortcuts?

**Comment by barthalion** [comment]
I get the same error from CLI, but launches fine with the desktop file somehow.

**Comment by rafatosta** [comment]
> Tem certeza de que iniciou o caminho certo através dos atalhos?



> Are you sure you launched the right one through the shortcuts?

Yes, everything is OK.

**Comment by rafatosta** [comment]
I have no idea why the error, but from the log it is related to the sip or package modules. I will investigate and resolve later today.

**Comment by rafatosta** [comment]
> I get the same error from CLI, but launches fine with the desktop file somehow.

Unfortunately I couldn't identify the problem. I compile and run normally by CLI the command:
flatpak-builder --run build com.rtosta.zapzap.yaml zapzap

Directly with:
flatpak run com.rtosta.zapzap
I couldn't figure out what the problem is

**Comment by hfiguiere** [comment]
> > I get the same error from CLI, but launches fine with the desktop file somehow.
> 
> Unfortunately I couldn't identify the problem. I compile and run normally by CLI the command: flatpak-builder --run build com.rtosta.zapzap.yaml zapzap

This is not running like the flatpak. This is running in a build environment.

Basically it seems that `packaging` is in the SDK but not in the runtime.

File: com.rtosta.zapzap.yaml [hunk 3]
```diff
@@ -0,0 +1,36 @@
+app-id: com.rtosta.zapzap
+runtime: org.kde.Platform
+runtime-version: "6.2"
+sdk: org.kde.Sdk
+base: io.qt.qtwebengine.BaseApp
+base-version: "6.2"
+comand: "zapzap"
```
Comment (hfiguiere) [inline com.rtosta.zapzap.yaml:7]:
```suggestion
command: "zapzap"
```

There is warning in the build about this non existent property (see typo)

And I'm wondering if that's not the source of all the problem.

**Review by hfiguiere — CHANGES_REQUESTED** [review]

**Comment by nedrichards** [comment]
bot, build com.rtosta.zapzap

**Pull request closed without merging**